app/controllers/placements/schools/placements/build_controller.rb

```diff
@@ -99,6 +110,19 @@ def update
         assign_additional_subjects
         render :add_additional_subjects and return
       end
+    when :add_year_group
+      year_group = params.dig(:placements_schools_placements_build_placement, :year_group)
+      @placement = build_placement
+      @placement.year_group = year_group
+
+      if @placement.valid_year_group?
+        session[:add_a_placement][:year_group] = year_group
+      else
+        @year_groups = Placements::Schools::Placements::Build::Placement.year_groups.map do |value, name|
```

Comment:
This is replicated code. Probably worth making a `year_groups` method and memoize the variable.